apps/vscode/src/host/hooks.ts

```diff
@@ -164,32 +165,18 @@ class EmbeddedStatementRangeProvider implements HostStatementRangeProvider {
     token: vscode.CancellationToken): Promise<hooks.StatementRange | undefined> {
     const vdoc = await virtualDoc(document, position, this._engine);
     if (vdoc) {
-      return await withVirtualDocUri(vdoc, document.uri, "statementRange", async (uri: vscode.Uri) => {
-        return getStatementRange(
-          uri,
-          adjustedPosition(vdoc.language, position),
-          vdoc.language
-        );
-      });
+      const result = await vscode.commands.executeCommand<hooks.StatementRange>(
+        "vscode.executeStatementRangeProvider",
+        document.uri,
+        adjustedPosition(vdoc.language, position)
+      );
+      return { range: unadjustedRange(vdoc.language, result.range), code: result.code };
```

Comment:
This set of changes in this file _breaks_ the statement range provider in source mode. We need to use the code as it was before the change, to propagate the statement range through the virtual doc. We can't call `vscode.executeStatementRangeProvider` on the `.qmd` but instead must call it on the vdoc.